### DISCUSSION TITLE
Some commands not available when using `sudo`

### DISCUSSION THREAD

**Discussion by emfrab** [pytest-dev/pytest#11197] [Q&A] [answered]
Hi everyone.
I'm having a weird issue with `pytest` on my Rocky Linux 9 machine. When I run it with `sudo`, some commands are not available, as shown below. In my case is particularly the `--teamcity` command that I need.

```
$ /bin/python -m pytest --help > help_no_sudo
$ sudo /bin/python -m pytest --help > help_sudo
$ diff help_no_sudo help_sudo
```
```diff
91,92d90
<   --teamcity            force output of JetBrains TeamCity service messages
<   --no-teamcity         disable output of JetBrains TeamCity service messages
193,195d190
< Custom options:
<   --jb-swapdiff         Swap actual/expected in diff
<
273,277d267
<   skippassedoutput (bool):
<                         skip output of passed tests for JetBrains TeamCity
<                         service messages
<   swapdiff (bool):      skip output of passed tests for JetBrains TeamCity
<                         service messages
```

As you can see, I used the full path for python executable, and the versions with and without `sudo` are the same:
```
$ /bin/python -m pytest --version
pytest 7.2.0
$ sudo /bin/python -m pytest --version
pytest 7.2.0
```

Any ideas? Is this by design?

**Comment by emfrab** [comment, marked as answer]
So the issue is that the plugin `pytest-teamcity` is not installed for `root` user.

**Reply by The-Compiler** [reply]
For what it's worth, I'd try to avoid running pytest as root in the first place - that sounds like quite a footgun...

**Reply by emfrab** [reply]
yeah I know, but most of the tests require root privileges to succeed. They are related to an external application which is dragging a poor design that requires elevated privileges to run.
Thank you!
